### PR TITLE
Maintenance: Replace window.location.host with env variable

### LINKED ISSUE #1766: Replace `window.location.host`
There should be a new environment variable added to store the env-specific app URL. It should replace all `window.location.host` calls in the code.

### PR DESCRIPTION
**Pull request by iamsamgibbs**
## Description

Added new env variable for URL to replace the use of window.location.host and window.location.origin

New APP_URL global const variable added as URL constructor, so `window.location.host` and `window.location.origin` can be replaced with `APP_URL.host` / `APP_URL.origin`

Updates the places they were used and added the variable to the docker file.

## Testing

Add URL variable to your .env file and build.

```
URL="http://localhost:9091"
```

## Diffs

**Changes** 🏗

* window.location.host and window.location.origin replaced with URL env variable
* New APP_URL global const variable added

Resolves #1766

### CODE REVIEW

File: src/redux/sagas/utils/getNetworkClient.ts
Comment (rdig) [inline src/redux/sagas/utils/getNetworkClient.ts:28]: This actually not a good change, since we were using `window.location.origin` as a fallback, which is still valid.

You still have to pass in a env variable `REPUTATION_ORACLE_ENDPOINT` to it, so if one's not set, chances are, the second is not set either.

Also, this is just a dev only use

File: src/routes/routeConstants.ts
Comment (rdig) [inline src/routes/routeConstants.ts:59]: This should be removed entirely, and just use `process.env.HOST` directly, not set inside a `const` and without a fallback

File: src/auth/index.ts
Comment (rdig) [inline src/auth/index.ts:9]: Same in this file as well, please remove the fallbacks as they're not needed anymore

File: .env.example
Comment (rdig) [inline .env.example:9]: I don't think that the origin is actually needed. I think we can just get away with using just the host.

The communication protocol should still come from the client, `window.location.protocol`, to which we just append the host

**Review by rdig — CHANGES_REQUESTED** [review]
Mostly good, however I left some notes which I want you to address first.

Thank you for putting this in

**Comment by iamsamgibbs** [comment]
Thank you for the review! I've fixed all those requests. :)

**Review by jakubcolony — CHANGES_REQUESTED** [review]
It seems like the majority of use cases require a full URL which leads us to having to combine the protocol and the host. 

How about storing app URL in env variable? 
- It can be used in links directly
- It's dead easy to get host from it:
```ts
const host = new URL(process.env.URL).host
```

What are your thoughts? This is also a question to @rdig I suppose

**Comment by rdig** [comment]
Actually yeah, that's a really smart idea! :100: 

Screenshot from 2024-02-07 13-58-01

Just have the `URL` as an env variable _(`http://localhost:9091` for local dev, `https://app.colony.io` for prod)_

Then put: `const appUrl = new URL(process.env.URL)` inside the global `constants` file, and just import it whenever you need it.

You can get whatever you need out it. Origin, Host, Protocolo, etc

Really good pickup @jakubcolony

**Comment by iamsamgibbs** [comment]
Great solution @jakubcolony - I've made these changes

**Review by rdig — APPROVED** [review]
All good now. Thank you Sam for putting up with all the back and fourth here!

Screenshot from 2024-02-07 20-34-06

One thing though, before merging, or rather just before pressing the merge button, make sure to do an announcement in the dev channel, to both our dev team, pagepro and the product team (since they run the app locally as well) to tell them that they need to update the `env` file

So as to not get a lot of people complaining about their local environment being broken

Another thought. After you merge this, QA will be broken until we introduce the new var to the online deployments, which we can't really do until @bogdan-1337 comes back.

So maybe wait until next week to merge this? Maybe?

**Review by jakubcolony — DISMISSED** [review]
LGTM 👍

**Review by jakubcolony — APPROVED** [review]

**Comment by rdig** [comment]
@iamsamgibbs heads up that @bogdan-1337 has now introduced the `URL` env variable on both QA and production, so that means that this can now be merged.

Do notify everyone when merging this, about updating they're respective `env` files

**Review by rdig — APPROVED** [review]

**Pull request merged**
